packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -890,6 +911,23 @@ describe('SignatureController', () => {
         ).version,
       ).toBe(SignTypedDataVersion.V3);
     });
+
+    // eslint-disable-next-line jest/no-disabled-tests
+    it.skip('invoke decoding api for permits', async () => {
+      const { controller } = createController();
+
+      await controller.newUnsignedTypedMessage(
+        PERMIT_PARAMS_MOCK,
+        PERMIT_REQUEST_MOCK,
+        SignTypedDataVersion.V4,
+        { parseJsonData: false },
+      );
+
+      await flushPromises();
+      expect(controller.state.signatureRequests[ID_MOCK].decodedRequest).toBe(
+        'IN_PROGRESS',
+      );
+    });
```

Comment:
I could not get this test to work, any suggestion here will be useful.

Reply:
Is the issue that we're awaiting `newUnsignedTypeMessage` but that won't resolve unless we mock the keyring controller?

Reply:
There are other similar tests here which are working, I think. reason is async call to decoding api.

Reply:
This is now fixed using mocking.